src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -1386,6 +1532,40 @@ public IList<string> ValidValues
         {
             get
             {
+                if (validValuesGeneratorType != null)
+                {
+                    ValidValuesGeneratorCacheEntry ValidValuesGeneratorCacheEntry;
+                    if (s_ValidValuesGeneratorCache.TryGetValue(validValuesGeneratorType, out ValidValuesGeneratorCacheEntry))
+                    {
+                        // The valid values generator is in the cache.
+                        // Update last access time. We use this for cache cleanup.
+                        var currentTime = DateTime.Now;
+                        ValidValuesGeneratorCacheEntry.cacheEntryAccessTime = currentTime;
+                        s_ValidValuesGeneratorCache[validValuesGeneratorType] = ValidValuesGeneratorCacheEntry;
+                    }
+                    else
+                    {
+                        // Add a valid values generator to the cache.
+                        // We don't cache valid values.
+                        // We expect that valid values can be cached in the valid values generator.
+                        var validValuesGenerator = (IValidateSetValuesGenerator)Activator.CreateInstance(validValuesGeneratorType);
+                        var cacheEntryAccessTime = DateTime.Now;
```

Comment:
Yes, now the cache is removed and we can create the generator in the constructor.
Fixed.